lib/src/package_analysis.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/context_builder.dart';
+import 'package:analyzer/dart/analysis/context_locator.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:args/args.dart';
+import 'package:path/path.dart' as path;
+
+Future<void> main(List<String> arguments) async {
+  // parse command-line arguments
+  final argResults = ArgParser().parse(arguments);
+  final paths = argResults.rest;
+
+  if (paths.length != 1) {
+    _exitFatal('Please specify a exactly one directory for analysis.', 1);
+  }
+
+  final packageLocation = paths.first;
+  if (!await FileSystemEntity.isDirectory(paths.first)) {
+    _exitFatal('Please specify a directory for analysis.', 1);
+  }
+
+  // initialise analyser
+  final context = ContextBuilder().createContext(
+    contextRoot: ContextLocator().locateRoots(
+      // use canonicalize to ensure we get an absolute path
+      includedPaths: [path.canonicalize(packageLocation)],
+    ).first,
+  );
+
+  for (final filePath in context.contextRoot.analyzedFiles()) {
+    final session = context.currentSession;
+    final someResolvedLibrary = await session.getResolvedLibrary(filePath);
+
+    // match [packageLocation]/lib/*.dart
```

Comment:
There are 3 kinds of libraries in a package which can be imported:
1. `lib/my_package.dart` (we usually import this library when we want to use the package)
2. `lib/*.dart` (could be in a sub-folder of `lib`, eg `lib/feature/*.dart`)
3. `lib/src/*.dart` (not in a sub-folder)

Even type 3 libraries can be imported for use, but it might make sense to exclude these libraries from analysis (by default, adding a switch to disable this).

Reply:
Yeah - anything under lib/src (also sub-folders) is considered private to the package by convention, and should not be included in the summary of the exported API surface.

https://dart.dev/guides/libraries/create-library-packages#what-makes-a-library-package